### PR TITLE
[DE] Add HassGetDate and HassGetTime

### PR DESCRIPTION
**Pull request by andreasbrett**
Added prefix "Es ist" for both GetDate and GetTime responses. Also added the weekday to GetDate responses as @TheFes mentioned in https://github.com/home-assistant/intents/pull/2309. Sounds much more natural to me and solves the ancient misunderstanding when asking...

- "Welcher Tag ist heute?"
- "Es ist der 13. Juli 2024"
- "Nein, ich meine den Wochentag"
- "Oh, okay, es ist Freitag"

## Summary by CodeRabbit

- **New Features**
  - Added German language support for querying the current date and time in Home Assistant.
  - Introduced response templates for the current date and time in German.

- **Tests**
  - Added German language test cases for querying the current date and time.

### CODE REVIEW

File: sentences/de/homeassistant_HassGetCurrentTime.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: de
+intents:
+  HassGetCurrentTime:
+    data:
+      - sentences:
+          - "wie spät ist es"
+          - "wie viel uhr (ist es[ jetzt]|haben wir)"
```
Comment (mib1185) [inline sentences/de/homeassistant_HassGetCurrentTime.yaml:7]:
```suggestion
          - "wie viel uhr (ist es|haben wir)[ jetzt]"
```
I think in some areas "wie viel uhr haben wir jetzt" is also a valid question.

File: sentences/de/homeassistant_HassGetCurrentDate.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+language: de
+intents:
+  HassGetCurrentDate:
+    data:
+      - sentences:
+          - "<welche> (tag|datum) (ist|haben wir) heute"
+          - "(nenn[e]|gib|gebe) mir das[ heutige] datum"
```
Comment (mib1185) [inline sentences/de/homeassistant_HassGetCurrentDate.yaml:7]:
```suggestion
          - "(nenn[e]|gib|gebe) mir das[ (heutige|aktuelle)] datum"
```
we should also allow "gebe mir das aktuelle datum"

**Review by mib1185 — CHANGES_REQUESTED** [review]
Hi @andreasbrett 
many thanks for this, really a great job :+1: 
I've just two small additions, which we should also add (_inlc. test sentence_), thx :+1:

**Comment by andreasbrett** [comment]
> Hi @andreasbrett many thanks for this, really a great job 👍 I've just two small additions, which we should also add (_inlc. test sentence_), thx 👍

Agreed, these changes make sense.

I noticed that STT (HA cloud in my case) translates the spoken "**wie viel** uhr ist es" to "**wieviel** uhr ist es" (see: https://www.korrekturen.de/wortliste/wie_viel.shtml), so I added that variant.

**Review by mib1185 — APPROVED** [review]
as always good job, your contributions are always highly appropriated @andreasbrett 👍

**Pull request merged**